tests/test-mongoc-collection.c

```diff
@@ -637,24 +637,57 @@ test_index_geo (void)
    mongoc_client_destroy(client);
 }
 
+static char *
+storage_engine (mongoc_client_t *client)
+{
+   bson_iter_t iter;
+   bson_error_t error;
+   bson_t cmd = BSON_INITIALIZER;
+   bson_t reply;
+   bool r;
+
+   /* NOTE: this default will change eventually */
+   char *engine = bson_strdup("mmapv1");
+
+   BSON_APPEND_INT32 (&cmd, "getCmdLineOpts", 1);
+   r = mongoc_client_command_simple(client, "admin", &cmd, NULL, &reply, &error);
+   ASSERT (r);
+
+   if (bson_iter_init_find (&iter, &reply, "parsed.storage.engine")) {
+      engine = bson_strdup(bson_iter_utf8(&iter, NULL));
+   }
+
+   bson_destroy (&reply);
+   bson_destroy (&cmd);
+
+   return engine;
+}
+
 static void
 test_index_storage (void)
 {
-   mongoc_collection_t *collection;
-   mongoc_database_t *database;
-   mongoc_client_t *client;
+   mongoc_collection_t *collection = NULL;
+   mongoc_database_t *database = NULL;
+   mongoc_client_t *client = NULL;
    mongoc_index_opt_t opt;
    mongoc_index_opt_wt_t wt_opt;
    bson_error_t error;
    bool r;
    bson_t keys;
-
-   mongoc_index_opt_init (&opt);
-   mongoc_index_opt_wt_init (&wt_opt);
+   char *engine = NULL;
 
    client = mongoc_client_new (gTestUri);
    ASSERT (client);
 
+   /* Skip unless we are on WT */
+   engine = storage_engine(client);
+   if (!engine || strcmp("wiredTiger", engine) != 0) {
```

Comment:
Well, if ```bson_iter_utf8(&iter, NULL)``` within storage_engine returned NULL (with properly-formatted bson input I don't think it would) then we would get NULL from that method.  Better safe than sorry on that front?